resources/src/androidMain/kotlin/dev/icerock/moko/resources/desc/StringDesc.kt

```diff
@@ -18,13 +18,32 @@ actual interface StringDesc {
         }
 
         actual class Custom actual constructor(
-            locale: String
+            private val locale: String
         ) : LocaleType() {
-            override val systemLocale: Locale = Locale(locale)
+            override val systemLocale: Locale get() {
+                val languageTagParts = locale.split("-")
+                return when (languageTagParts.size) {
+                    LANGUAGE -> Locale(languageTagParts[0])
+                    LANGUAGE_AND_COUNTRY -> Locale(languageTagParts[0], languageTagParts[1])
+                    LANGUAGE_AND_COUNTRY_AND_VARIANT -> Locale(
+                        languageTagParts[0],
+                        languageTagParts[1],
+                        languageTagParts[2]
+                    )
+                    else -> throw IllegalArgumentException(
+                        "Invalid language tag $locale which has more than three parts."
+                    )
+                }
+            }
         }
     }
 
     actual companion object {
         actual var localeType: LocaleType = LocaleType.System
     }
 }
+
+private const val LANGUAGE = 1
+private const val LANGUAGE_AND_COUNTRY = 2
+private const val LANGUAGE_AND_COUNTRY_AND_VARIANT = 3
```

Comment:
This is silly but I had to do it to get around Detekt.

Reply:
if we put it to `actual companion object` we can remove `@file:JvmName("StringDescJvm")` and not broke ABI